src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDatabase.java

```diff
@@ -102,7 +101,15 @@ private void migrateDB(Flyway flyway) throws IOException, FlywayException {
 
     private void recreateDB(Flyway flyway) throws IOException {
         logger.atDebug().kv("database-path", databasePath.toString()).log("Deleting the existing DB");
-        Files.deleteIfExists(databasePath);
+        Files.list(databasePath).forEach((path -> {
```

Comment:
files.list gives you a stream that you MUST close. Use this with try-with-resources